### ISSUE TITLE
[Doctrine Bridge] Length Constraint on simple_array fields

### ISSUE THREAD

**Issue by belinde** [symfony/symfony#31855]
**Symfony version(s) affected**: 4.3.0

**Description**  
When using a `simple_array` field, a `Length` constraint is always applied if the doctrine mapping contains a `length` value. However, the valued checked by the constraint is an array, so the constraint itself fails. I think that not every doctrine type must be checked by length, even if it's limited on database: I personally use often the type `json_array` with small length in order to have SMALLTEXT instead of LONGTEXT.

**How to reproduce**  
Make an entity with a `simple_array` field, make a form with validation `Choice` with `multiple=true`, and try to validate the form. The error "This value should be of type string." is shown, even if the submitted data is an array with valid values

**Possible Solution**  
In `Validator\DoctrineLoader::loadClassMetadata()`, in place of this code:
```php
if (null === ($mapping['length'] ?? null)) {
    continue;
}
```
we should check also on `$mapping['type']` and skip if the type isn't a real string type. I'm just not sure about what types we should check, and worst how to manage custom types: probably is safe to skip the check.

As a side note: this validation broke my installation (also in production, as the form using it is not so commoly used and is not covered by test; my fault), and I really can't find documentation about the feature at all: I see that I can skip some entities through regexp, but I can't understand where to put such configuration.

**Comment by xabbuh** [comment]
The fact that this was automatically enabled for all classes was a bug which will be fixed in 4.3.1 where you really need to opt-in for the auto validation feature.

Something like this should also work:

```yaml
framework:
    validation:
        auto_mapping:
            App\: []
```

Apart from that I agree that blindly applying the `Length` constraint is a bug.

**Comment by belinde** [comment]
The workaround you proposed didn't work for me; I tried to to this:
```yaml
  validation:
    auto_mapping:
      JustSomeWorkaround\: []
```
In this way the regexp always fail, so no entity check is done altogheter; I'll just wait for 4.3.1 to remove it.

I also checked the [official Doctrine documentation](https://www.doctrine-project.org/projects/doctrine-dbal/en/2.9/reference/types.html) and I think it's safe do the `Length` check only on types `string`, `text` and `guid`. For other basic types it doesn't make any sense, and custom types are a "must not touch" for me.

Is anyone working on this issue? Just the check on string types for `Length`, not the opt-in thing. If not, I can provide a PR.
